### PR TITLE
web-handlers: support for presigned-get json-rpc call for MinioBrowser's "share" feature.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
@harshavardhana implementing presigned-get in minio-browser is more complicated than implementing it in minio-server hence done this way.

### CODE REVIEW

File: cmd/signature-v4.go [hunk 1]
```diff
@@ -417,3 +419,36 @@ func doesSignatureMatch(hashedPayload string, r *http.Request, validateRegion bo
...
+		"X-Amz-Algorithm=" + signV4Algorithm,
+		"X-Amz-Credential=" + strings.Replace(credential, "/", "%2F", -1),
+		"X-Amz-Date=" + dateStr,
+		"X-Amz-Expires=" + "604800",
+		"X-Amz-SignedHeaders=host",
+	}, "&")
+
+	path := "/" + path.Join(bucket, object)
+
+	var extractedSignedHeaders http.Header // will be nil as "host" is the only signed header.
```
Comment (krisis) [inline cmd/signature-v4.go:446]: I couldn't understand the comment about `host`  being the only signed header. Could you explain that?

**Review by krisis — COMMENTED** [review]
Besides the question I have asked on the comment, rest of the changes LGTM.

**Comment by harshavardhana** [comment]
> I couldn't understand the comment about host being the only signed header. Could you explain that?

Host is the only header that is needed at least for presigned URLs to work.

**Comment by harshavardhana** [comment]
@krishnasrinivas please add a test case.

**Comment by harshavardhana** [comment]
> @harshavardhana implementing presigned-get in minio-browser is more complicated than implementing it in minio-server hence done this way.

This looks good.

**Pull request closed without merging**